### PR TITLE
Test we get back out the data we put in.

### PR DESCRIPTION
**Pull request by brian-brazil**
This also adds a utility that'll aid future tests.

### CODE REVIEW

File: db_test.go [hunk 1]
```diff
@@ -16,11 +16,37 @@ package tsdb
 import (
 	"io/ioutil"
 	"os"
+	"reflect"
 	"testing"
 
 	"github.com/prometheus/tsdb/labels"
 )
 
+type Sample struct {
```
Comment (fabxc) [inline db_test.go]: Make type and attributes unexported.

File: db_test.go [hunk 2]
```diff
@@ -16,11 +16,37 @@ package tsdb
...
+		samples := []Sample{}
+		it := series.Iterator()
+		for it.Next() {
+			t, v := it.At()
+			samples = append(samples, Sample{T: t, V: v})
+		}
+
+		name := series.Labels().String()
+		result[name] = samples
+	}
```
Comment (fabxc) [inline db_test.go]: This has to end with `if err := ss.Err(); ...`

File: db_test.go [hunk 3]
```diff
@@ -52,9 +78,12 @@ func TestDataAvailableOnlyAfterCommit(t *testing.T) {
...
 	defer querier.Close()
 
-	seriesSet = querier.Select(labels.NewEqualMatcher("foo", "bar"))
-	if !seriesSet.Next() {
-		t.Fatalf("Error, was expecting a matching series.")
+	seriesSet = readSeriesSet(querier.Select(labels.NewEqualMatcher("foo", "bar")))
+	expected := map[string][]Sample{
+		`{foo="bar"}`: []Sample{{T: 0, V: 0}},
+	}
+	if !reflect.DeepEqual(seriesSet, expected) {
```
Comment (fabxc) [inline db_test.go]: So we are using the `require` package in a bunch of places in our projects by now.
Many people advocate against only using stdlib stuff, but practically, it does exactly the same and is less noisy. I think we should use it somewhat consistently.

File: db_test.go (hunk 1 above)
Comment (brian-brazil) [inline db_test.go]: Done

File: db_test.go (hunk 3 above)
Comment (brian-brazil) [inline db_test.go]: I don't see a big problem with using it in tests, the client library is the only place I'd worry about deps.

**Comment by brian-brazil** [comment]
Updated, PTAL.

**Pull request merged**